pkg/provider/provider.go

```diff
@@ -532,6 +559,20 @@ func (k *kubeProvider) gvkFromURN(urn resource.URN) schema.GroupVersionKind {
 	}
 }
 
+func (k *kubeProvider) readLiveObject(
+	obj *unstructured.Unstructured,
+) (*unstructured.Unstructured, error) {
+	// Retrieve live version of last submitted version of object.
+	clientForResource, err := client.FromResource(k.pool, k.client, obj)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get the "live" version of the last submitted object. This is necessary because the server may
+	// have populated some fields automatically, updated status fields, and so on.
+	return clientForResource.Get(obj.GetName(), metav1.GetOptions{})
+}
+
```

Comment:
Does it make sense for the implementation of `Get` that has a copy of this code to now call this function?

Reply:
If I understand the suggestion, I think it's not worth sharing since (1) the code is pretty trivial and this does not seem to add to the clarity IMO, and (2) in the other scenarios we are passed the client from the client pool, so obtaining it again is kind of a waste?